src/extra/stream/client.ts

```diff
@@ -83,6 +86,84 @@ export class StreamClient extends EventDispatcher implements Stream {
     }
   }
 
+  /**
+   * Get the available livestream resolutions for a camera component. If the
+   * stream client cannot find any available resolutions, an empty list will be
+   * returned.
+   *
+   * @param resourceName - The name of a camera component.
+   * @returns A list of available resolutions for livestreaming.
+   */
+  async getOptions(resourceName: string): Promise<Resolution[]> {
+    const fetchOptions = async (name: string): Promise<Resolution[]> => {
+      const request = new GetStreamOptionsRequest({ name });
+      this.options.requestLogger?.(request);
+      try {
+        const response = await this.client.getStreamOptions(request);
+        return response.resolutions;
+      } catch {
+        return [];
```

Comment:
Why return an empty list when the available resolution is not available and not an error message?

Reply:
Yeah good question! I toyed around with erroring out in the failure case but thought it would be nicer to just return empty array to avoid complexity when calling it in the frontend and trying to fill a dropdown or such with the return. (This is presumptuous as I do not have that much experience with frontend stuff) Happy to change the behavior to throw if that is better.

Reply:
I think simple is better! keep how it is